src/core/document.js

```diff
@@ -793,6 +793,7 @@ class Page {
           return [];
         }
 
+        const orphanFields = (await this.fieldObjects)?.orphanFields;
```

Comment:
I'm not convinced that this is generally safe unfortunately, and it'd also be nice to avoid unnecessarily delaying annotation-parsing here since most documents probably won't have any fieldObjects.

```diff
diff --git a/src/core/document.js b/src/core/document.js
index 68e26e8a2..d5c453d5a 100644
--- a/src/core/document.js
+++ b/src/core/document.js
@@ -787,13 +787,15 @@ class Page {
         if (annots.length === 0) {
           return annots;
         }
-        const annotationGlobals =
-          await this.pdfManager.ensureDoc("annotationGlobals");
+        const [annotationGlobals, fieldObjects] = await Promise.all([
+          this.pdfManager.ensureDoc("annotationGlobals"),
+          this.pdfManager.ensureDoc("fieldObjects"),
+        ]);
         if (!annotationGlobals) {
           return [];
         }

-        const orphanFields = (await this.fieldObjects)?.orphanFields;
+        const orphanFields = fieldObjects?.orphanFields;
         const annotationPromises = [];
         for (const annotationRef of annots) {
           annotationPromises.push(
```